File: src/main/java/graph/algorithm/Algorithm.java
package graph.algorithm;

import graph.vertex.Vertex;
import org.jetbrains.annotations.NotNull;

import java.util.HashMap;

public interface Algorithm {
    Integer INFINITY = Integer.MAX_VALUE;

    boolean isAccessible(@NotNull Vertex from, @NotNull Vertex to);
    int getMinDistance(@NotNull Vertex from, @NotNull Vertex to);
    HashMap<Vertex, Integer> getDistances(@NotNull Vertex node);
    GraphPath getPath(@NotNull Vertex from, @NotNull Vertex to);
}

File: src/main/java/mvc/model/algorithms/Algorithm.kt
package mvc.model.algorithms

import mvc.model.node.Node

import java.util.HashMap

interface Algorithm {
    fun isAccessible(from: Node, to: Node): Boolean
    fun getMinDistance(from: Node, to: Node): Int
    fun getDistances(node: Node): HashMap<Node, Int>
    fun getPath(from: Node, to: Node): GraphPath

    companion object {
        const val INFINITY = Integer.MAX_VALUE
    }
}
